modules/proto/core/src/smithyproto/proto3/Compiler.scala

```diff
@@ -67,30 +67,7 @@ class Compiler() {
       excludeInternal(s) || Prelude.isPreludeShape(s) || traitShapes(s)
   }
 
-  /** Unused union shape are not exported.
-    *
-    * Union shapes used exactly once are exported within the structure that uses
-    * them.
-    *
-    * If they're used more than once, this function will throw.
-    */
-  private def validateUnionShapes(model: Model): Unit = {
```

Comment:
I agree that we should change the logic to allow for unions to be in several messages. 

However, you can't just decided to reify messages to hold the union field. The impact is too important. 

I think the more correct course of action would just be to remove the validation and to inline the union as a `oneOf` in all structures that use it.